### PR TITLE
[Refactor] Support to use "split" to specify training set/validation set in the ImageNet dataset

### PR DESCRIPTION
**Pull request by zzc98**
## Motivation

- Support to use `split` to specify training set/validation set in the ImageNet dataset
- Support to use `split` to specify training set/test set in the Mnist/Fashion-Mnist dataset
- Support to use `split` to specify training set/validation set in the VOC dataset
- Refactor SUN397 datasets
- Update `docs/en/user_guides/dataset_prepare.md` and `docs/zh_CN/user_guides/dataset_prepare.md`

## Examples

```python
# ImageNet
>>> from mmpretrain.datasets import ImageNet
>>> train_dataset = ImageNet(data_root='data/imagenet', split='train')
>>> train_dataset
Dataset ImageNet
	Number of samples:  1281167
	Number of categories:       1000
	Root of dataset:    data/imagenet
>>> test_dataset = ImageNet(data_root='data/imagenet', split='val')
>>> test_dataset
Dataset ImageNet
	Number of samples:  50000
	Number of categories:       1000
	Root of dataset:    data/imagenet

# ImageNet21K
>>> from mmpretrain.datasets import ImageNet21k
>>> train_dataset = ImageNet21k(data_root='data/imagenet21k', split='train')
>>> train_dataset
Dataset ImageNet21k
	Number of samples:  14197088
	Annotation file:    data/imagenet21k/meta/train.txt
	Prefix of images:   data/imagenet21k/train
    
# VOC
>>> from mmpretrain.datasets import VOC
>>> train_dataset = VOC(data_root='data/VOC2007', split='trainval')
>>> train_dataset
Dataset VOC
	Number of samples:  5011
	Number of categories:       20
	Prefix of dataset:  data/VOC2007
	Path of image set:  data/VOC2007/ImageSets/Main/trainval.txt
	Prefix of images:   data/VOC2007/JPEGImages
	Prefix of annotations:      data/VOC2007/Annotations
>>> test_dataset = VOC(data_root='data/VOC2007', split='test')
>>> test_dataset
Dataset VOC
	Number of samples:  4952
	Number of categories:       20
	Prefix of dataset:  data/VOC2007
	Path of image set:  data/VOC2007/ImageSets/Main/test.txt
	Prefix of images:   data/VOC2007/JPEGImages
	Prefix of annotations:      data/VOC2007/Annotations

# SUN397
>>> from mmpretrain.datasets import SUN397
>>> train_dataset = SUN397(data_root='data/SUN397', split='train')
>>> train_dataset
Dataset SUN397
	Number of samples:  19850
	Number of categories:       397
	Root of dataset:    data/SUN397
>>> test_dataset = SUN397(data_root='data/SUN397', split='test')
>>> test_dataset
Dataset SUN397
	Number of samples:  19850
	Number of categories:       397
	Root of dataset:    data/SUN397
```

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpretrain/datasets/imagenet.py
Comment (fangyixiao18) [inline mmpretrain/datasets/imagenet.py:1]: please also refactor `ImageNet21k`

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: docs/en/user_guides/dataset_prepare.md
Comment (fangyixiao18) [inline docs/en/user_guides/dataset_prepare.md:1]: please also update Chinese version

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

**Review by mzr1996 — APPROVED** [review]

**Review by fangyixiao18 — APPROVED** [review]
